### PR TITLE
Enable building PowerShell for Apple M1 runtime

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Add support to `Start-PSBuild` with runtime `osx-arm64`.  Requires https://github.com/PowerShell/PowerShell-Native/pull/64 to be published first.  A local build of both works.  When looking for crossgen, there appears to be one found under `x64-arm64` folder which fails, but the one in the parent `tools` folder works so need to filter that out for when finding crossgen.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by jborean93** [comment]
So does this build a universal binary or just one compiled for arm64? Does it require any work from .NET, I remember reading something about M1 support in the new preview so assumed PowerShell had to wait for that.

**Comment by SteveL-MSFT** [comment]
@jborean93 This PR leverages dotnet to build M1 optimized binaries.  It depends on https://github.com/PowerShell/PowerShell-Native/pull/64 which builds universal native binaries for pwsh.

**Comment by jborean93** [comment]
Awesome thanks for the info, will have to get a move on updating the mi fork to produce arm64 libraries so powershell can use them.

File: build.psm1 [hunk 1]
```diff
@@ -2386,6 +2394,7 @@ function Start-CrossGen {
     $crossGenPath = Get-ChildItem $script:Environment.nugetPackagesRoot $crossGenExe -Recurse | `
                         Where-Object { $_.FullName -match $crossGenRuntime } | `
                         Where-Object { $_.FullName -match $dotnetRuntimeVersion } | `
+                        Where-Object { (Split-Path $_.FullName -Parent).EndsWith('tools') } | `
```
Comment (PaulHigin) [inline build.psm1:2397]: Is this 'tools' path filter just for macOS, or for all runtimes?  I assume the latter, but it is not clear to me.  Maybe add a comment to reassure.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Would you submit another PR to update the release build or add that to this PR?

File: build.psm1 (hunk 1 above)
Comment (SteveL-MSFT) [reply build.psm1:2397]: The `tools` folder is for all runtimes, but for some reason, x64-arm64 for macOS has a `crossgen` in that additional runtime folder which is discovered, but doesn't actually work.  The one in `tools` does work.

**Comment by SteveL-MSFT** [comment]
@TravisEz13 I can update this PR to also do the build, was using this to test locally

**Comment by SteveL-MSFT** [comment]
@TravisEz13 I spent some time looking at the yml and psm1 that does the build, but it's much more complicated now than the last time I made changes.  I would suggest having it integrated with the build as a separate PR.

**Review by PaulHigin — APPROVED** [review]

**Comment by TravisEz13** [comment]
@SteveL-MSFT  is this ready to merge?  Don't we need the updated native nuget?

**Comment by adityapatwardhan** [comment]
The description says it is dependent on the PR in PS-Native repo. The NuGet package after merging that that PR has to be released too.

**Comment by SteveL-MSFT** [comment]
For the full package to work, it requires the universal binary from powershell-native, however, this PR itself can be merged.  @anmenaga can you do a separate PR to enable building of the M1 pkg?

**Pull request merged**